Add "Access Graph" docs

jakule opened the pull request:
Access Graph docs have been added under slug /access-graph/. This document provides details about the Teleport Access Graph. Currently, this feature is only available to Teleport Cloud Team users and is in its preview stage.

Closes https://github.com/gravitational/access-graph/issues/284

jakule committed:
Add "Access Graph" docs

Access Graph docs have been added under slug /access-graph/. This document provides details about the Teleport Access Graph. Currently, this feature is only available to Teleport Cloud Team users and is in its preview stage.

Signed-off-by: Jakub Nyckowski <jakub.nyckowski@goteleport.com>

jakule committed:
Add images

Signed-off-by: Jakub Nyckowski <jakub.nyckowski@goteleport.com>

jakule committed:
Fix typo

Signed-off-by: Jakub Nyckowski <jakub.nyckowski@goteleport.com>

ptgott reviewed (commented):
For me, this is midway between a blog post and a documentation page. 

As documentation, the page introduces major terms (e.g., users, user groups, and actions), and instructs the user to visit the Web UI. The "How to use it" section does not include instructions, but rather suggests use cases that the user can discover on their own.

The first paragraph, "Teleport Access Graph is a new way to visualize and understand access to your infrastructure," sounds like an announcement rather than documentation.

Would it be more effective to make this a blog post instead of a docs page? (CC @travelton @benarent)

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,78 @@
+
+<Admonition type="tip" title="Preview">
+  Teleport Access Graph is currently in Preview. Currently, it is only available
+  to the Teleport Cloud Team users. Not all Teleport resources are currently
```
Does "Teleport Cloud Team" mean both "Teleport Enterprise Cloud" and "Teleport Team"?

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,78 @@
+<Admonition type="tip" title="Preview">
+  Teleport Access Graph is currently in Preview. Currently, it is only available
+  to the Teleport Cloud Team users. Not all Teleport resources are currently
+  supported.
```
Rather than saying that not all resources are supported, we should list the resources that are supported.

jakule commented:
> For me, this is midway between a blog post and a documentation page.
> 
> As documentation, the page introduces major terms (e.g., users, user groups, and actions), and instructs the user to visit the Web UI. The "How to use it" section does not include instructions, but rather suggests use cases that the user can discover on their own.
> 
> The first paragraph, "Teleport Access Graph is a new way to visualize and understand access to your infrastructure," sounds like an announcement rather than documentation.
> 
> Would it be more effective to make this a blog post instead of a docs page? (CC @travelton @benarent)

@ptgott I'm not saying that I disagree, but Access Graph will be enabled next week for our Cloud users. As a user, I'd assume that new features will have something in the product documentation. If we move this to our blog, we will have nothing to put in our docs. I can rephrase the docs, but I think that it's important to mention what it is, how you can use it, and what value this "new thing" gives you.

jakule committed:
Updated docs for Teleport Access Graph feature

Expanded documentation for Teleport Access Graph feature to include more detailed information about the usage and functionality. Additional visual aids were added to provide a clearer understanding of the concept. Also, specific instructions were included to spotlight the versatility of the feature for advanced user control and management.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Add detailed description of Graph paths in Teleport Access Graph doc

Included detailed explanations about the allow paths and deny paths in the Teleport Access Graph documentation. This change provides users with clearer insights about accessing and managing resources. Also added imagery to visually reinforce the concepts of the Access Graph.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Normalize the size images

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Update Access Graph documentation for clarity

This commit refines the Access Graph section of the Teleport documentation. It provides a clearer sequence of paths, detailing how they proceed from Users to Resources. Also, the description of how User Groups are created, either from Roles or accepted access requests, has been enhanced for more clarity. This aims to improve user understanding of Teleport functionalities.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,166 @@
+---
+
+<Admonition type="tip" title="Preview">
+  Teleport Access Graph is currently in Preview. Currently, it is only available
+  to the Teleport Cloud Team users. Currently, only SSH server are supported.
```
```suggestion
  Teleport Access Graph is currently in Preview. Currently, it is only available
  to Teleport Team users, and only visualizes access to SSH servers enrolled with your Teleport clusters.
```
Suggested to:
- Make this consistent with the rest of the docs, which say "Teleport Team", rather than "Teleport Cloud Team"
- Be more explicit about what "are supported" means. I'm not sure if my suggestion is technically correct, though.

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,166 @@
+
+Teleport Access Graph shows the relationships between users, roles, and
+resources. It does this by showing paths between nodes. Paths are the
+relationships between nodes. Paths always go from:
+
+  Users to User Groups to Actions to Resource Groups to Resources
```
```suggestion
relationships between nodes. Paths always connect nodes in the following order:

1. Users
1. User Groups
1. Actions
1. Resource Groups
1. Resources
```
Suggested to make this a little easier to read

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,166 @@
+
+#### Resource Groups
+
+Resource Groups are created from Teleport Roles.
```
```suggestion
Resource Groups are created from Teleport roles.
```

ptgott commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,166 @@
+#### Resources
+
+Resources are created from Teleport resources like nodes, databases, and
+Kubernetes clusters.
```
```suggestion
Kubernetes clusters. Currently, only SSH nodes are supported.
```
Is this what the Admonition at the top means? If so, I've suggested mentioning it here as well.

jakule committed:
Address review comments

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Update user import details in access-graph.mdx

This update adds important information about user importation in the Teleport documentation. It specifies that local users are imported as soon as they are created, while external users are only imported when they log in for the first time.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

r0mant commented on docs/pages/access-graph.mdx:
```diff
@@ -0,0 +1,172 @@
+
+## Graph nodes
+
+Teleport Access Graph divides your infrastructure into five main components:
```
> five main components

There's 8 below from what I can see.

jakule committed:
Refine explanation and structure of access-graph.mdx

The Teleport Access Graph documentation is updated to improve clarity and accuracy. It now explains more clearly how it visualizes access to infrastructure, expands the main components from five to eight, and refines the explanation on User Groups. Additionally, the Access Graph section is relocated within the documentation structure for better organization.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Refine explanation and structure of access-graph.mdx

The Teleport Access Graph documentation is updated to improve clarity and accuracy. It now explains more clearly how it visualizes access to infrastructure, expands the main components from five to eight, and refines the explanation on User Groups. Additionally, the Access Graph section is relocated within the documentation structure for better organization.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Refine explanation and structure of access-graph.mdx

The Teleport Access Graph documentation is updated to improve clarity and accuracy. It now explains more clearly how it visualizes access to infrastructure, expands the main components from five to eight, and refines the explanation on User Groups. Additionally, the Access Graph section is relocated within the documentation structure for better organization.

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>

jakule committed:
Update docs/pages/access-graph.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

jakule committed:
Merge branch 'master' into jakule/access-graph-docs

jakule committed:
Merge branch 'master' into jakule/access-graph-docs

jakule committed:
Fix linter errors

Signed-off-by: Jakub Nyckowski <nyckowski.jakub@gmail.com>
